plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -571,7 +572,7 @@ func (c *CloudWatch) BuildMetricDatum(point telegraf.Metric) []*cloudwatch.Metri
 // 30 dimensions per metric so we only keep up to the first 30 alphabetically.
 // This always includes the "host" tag if it exists.
 // See https://github.com/aws/amazon-cloudwatch-agent/issues/398
-func BuildDimensions(mTags map[string]string) []*cloudwatch.Dimension {
+func BuildDimensions(mTags map[string]string, globalDimensions map[string]string) []*cloudwatch.Dimension {
```

Comment:
```suggestion
func BuildDimensions(mTags , globalDimensions map[string]string) []*cloudwatch.Dimension {
```
Or would it better to pass the mTags and globalDimensions as a map which globalDimensions will override all the key from mTags. Then `sort.Strings(keys)`

Reply:
If this were my codebase, I'd definitely prefer explicitly typing each argument (as in the original PR). But it's not my codebase so it's up to you.

There are issues with combining the maps because:

- The `host` dimension from `mTags` needs to be added first
- The current code opts to prioritise the global tags over the metric tags when `MaxDimensions` is reached so we need to keep the two separate until we've made that check.

Reply:
I am opposed to combining the maps into one argument. I'm indifferent if we combine the types or not